superset/dashboards/commands/importers/v0.py

```diff
@@ -0,0 +1,336 @@
+        session.query(Slice).filter(Slice.id.in_(old_to_new_slc_id_dict.values())).all()
+    )
+
+    if existing_dashboard:
+        existing_dashboard.override(dashboard_to_import)
+        existing_dashboard.slices = new_slices
+        session.flush()
+        return existing_dashboard.id
+
+    dashboard_to_import.slices = new_slices
+    session.add(dashboard_to_import)
+    session.flush()
+    return dashboard_to_import.id  # type: ignore
+
+
+def decode_dashboards(  # pylint: disable=too-many-return-statements
+    o: Dict[str, Any]
+) -> Any:
+    """
+    Function to be passed into json.loads obj_hook parameter
+    Recreates the dashboard object from a json representation.
+    """
+    from superset.connectors.druid.models import (
+        DruidCluster,
+        DruidColumn,
+        DruidDatasource,
+        DruidMetric,
+    )
+
+    if "__Dashboard__" in o:
+        return Dashboard(**o["__Dashboard__"])
+    if "__Slice__" in o:
+        return Slice(**o["__Slice__"])
+    if "__TableColumn__" in o:
+        return TableColumn(**o["__TableColumn__"])
+    if "__SqlaTable__" in o:
+        return SqlaTable(**o["__SqlaTable__"])
+    if "__SqlMetric__" in o:
+        return SqlMetric(**o["__SqlMetric__"])
+    if "__DruidCluster__" in o:
+        return DruidCluster(**o["__DruidCluster__"])
+    if "__DruidColumn__" in o:
+        return DruidColumn(**o["__DruidColumn__"])
+    if "__DruidDatasource__" in o:
+        return DruidDatasource(**o["__DruidDatasource__"])
+    if "__DruidMetric__" in o:
+        return DruidMetric(**o["__DruidMetric__"])
+    if "__datetime__" in o:
+        return datetime.strptime(o["__datetime__"], "%Y-%m-%dT%H:%M:%S")
+
+    return o
+
+
+def import_dashboards(
+    session: Session,
+    content: str,
+    database_id: Optional[int] = None,
+    import_time: Optional[int] = None,
+) -> None:
+    """Imports dashboards from a stream to databases"""
+    current_tt = int(time.time())
+    import_time = current_tt if import_time is None else import_time
+    data = json.loads(content, object_hook=decode_dashboards)
+    if not data:
+        raise DashboardImportException(_("No data in file"))
+    for table in data["datasources"]:
+        type(table).import_obj(table, database_id, import_time=import_time)
+    session.commit()
+    for dashboard in data["dashboards"]:
+        import_dashboard(dashboard, import_time=import_time)
+    session.commit()
+
+
+class ImportDashboardsCommand(BaseCommand):
+    """
+    Import dashboard in JSON format.
+
+    This is the original unversioned format used to export and import dashboards
+    in Superset.
+    """
+
+    def __init__(self, contents: Dict[str, str], database_id: Optional[int] = None):
+        self.contents = contents
+        self.database_id = database_id
+
+    def run(self) -> None:
+        self.validate()
+
+        for file_name, content in self.contents.items():
+            logger.info(f"Importing dashboard from file {file_name}")
```

Comment:
dnt we want to use `%s` instead of fstring?

Reply:
Agh, good catch. For sure!

Reply:
I did this PR before the other one where I started using string interpolation.